cmd/ova-provider-server/nfs_test.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/onsi/gomega"
+)
+
+func TestFindOVAFiles(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	tests := []struct {
+		name         string
+		setup        func(directory string)
+		expectedOVAs []string
+		expectedOVFs []string
+		expectError  bool
+	}{
+		{
+			name: "basic structure",
+			setup: func(directory string) {
+				_ = ioutil.WriteFile(filepath.Join(directory, "test.ova"), []byte{}, 0644)
+				_ = ioutil.WriteFile(filepath.Join(directory, "test.ovf"), []byte{}, 0644)
+				_ = os.Mkdir(filepath.Join(directory, "subdir1"), 0755)
+				_ = ioutil.WriteFile(filepath.Join(directory, "subdir1", "test1.ova"), []byte{}, 0644)
+				_ = ioutil.WriteFile(filepath.Join(directory, "subdir1", "test1.ovf"), []byte{}, 0644)
+				_ = os.Mkdir(filepath.Join(directory, "subdir1", "subdir2"), 0755)
+				_ = ioutil.WriteFile(filepath.Join(directory, "subdir1", "subdir2", "test2.ovf"), []byte{}, 0644)
+			},
+			expectedOVAs: []string{"test.ova", "subdir1/test1.ova"},
+			expectedOVFs: []string{"test.ovf", "subdir1/test1.ovf", "subdir1/subdir2/test2.ovf"},
+			expectError:  false,
+		},
+		{
+			name: "non-existent directory",
+			setup: func(directory string) {
+				_ = os.RemoveAll(directory)
+			},
+			expectedOVAs: nil,
+			expectedOVFs: nil,
+			expectError:  true,
+		},
+		{
+			name: "non-ova/ovf files",
+			setup: func(directory string) {
+				_ = ioutil.WriteFile(filepath.Join(directory, "test.txt"), []byte{}, 0644)
+			},
+			expectedOVAs: nil,
+			expectedOVFs: nil,
+			expectError:  false,
+		},
+		{
+			name: "incorrect depth",
+			setup: func(directory string) {
+				_ = os.Mkdir(filepath.Join(directory, "subdir1"), 0755)
+				_ = os.Mkdir(filepath.Join(directory, "subdir1", "subdir2"), 0755)
+				_ = ioutil.WriteFile(filepath.Join(directory, "subdir1", "subdir2", "test3.ova"), []byte{}, 0644)
+			},
+			expectedOVAs: nil,
+			expectedOVFs: nil,
+			expectError:  false,
+		},
+		{
+			name: "folder with extantion",
+			setup: func(directory string) {
+				_ = os.Mkdir(filepath.Join(directory, "subdir1.ova"), 0755)
+				_ = os.Mkdir(filepath.Join(directory, "subdir2.ovf"), 0755)
+			},
+			expectedOVAs: nil,
+			expectedOVFs: nil,
```

Comment:
why don't we expect to get the ovf?

Reply:
ah, subdir2.ovf is a directory.. ok

Reply:
yeah its a directory, and good idea, i"ll add this case.